brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java

```diff
@@ -0,0 +1,41 @@
+package io.cloudsoft.tosca.a4c.brooklyn;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.apache.brooklyn.config.ConfigKey;
+
+import com.google.common.base.Optional;
+
+import alien4cloud.component.repository.exception.CSARVersionNotFoundException;
```

Comment:
wrt the use in `getCsarPath` you could either add a new exception type or change the method's return type to `Optional<Path>`, returning absent when no path is found for the arguments. The latter choice feels more consistent with the rest of the interface.